vm/src/obj/objgetset.rs

```diff
@@ -0,0 +1,230 @@
+    }
+}
+
+impl<F, S, R> IntoPyGetterFunc<(RefParam<S>, R)> for F
+where
+    F: Fn(&S) -> R + 'static,
+    S: PyValue,
+    R: IntoPyObject,
+{
+    fn into_getter(self) -> PyGetterFunc {
+        IntoPyGetterFunc::into_getter(move |zelf: &S, _vm: &VirtualMachine| (self)(zelf))
+    }
+}
+
+pub trait IntoPySetterFunc<T> {
+    fn into_setter(self) -> PySetterFunc;
+}
+
+impl<F, T, V, R> IntoPySetterFunc<(OwnedParam<T>, V, R, VirtualMachine)> for F
+where
+    F: Fn(T, V, &VirtualMachine) -> R + 'static,
+    T: TryFromObject,
+    V: TryFromObject,
+    R: IntoPySetResult,
+{
+    fn into_setter(self) -> PySetterFunc {
+        Box::new(move |obj, value, vm| {
+            let obj = T::try_from_object(vm, obj)?;
+            let value = V::try_from_object(vm, value)?;
+            (self)(obj, value, vm).into_pysetresult()
+        })
+    }
+}
+
+impl<F, S, V, R> IntoPySetterFunc<(RefParam<S>, V, R, VirtualMachine)> for F
+where
+    F: Fn(&S, V, &VirtualMachine) -> R + 'static,
+    S: PyValue,
+    V: TryFromObject,
+    R: IntoPySetResult,
+{
+    fn into_setter(self) -> PySetterFunc {
+        Box::new(move |obj, value, vm| {
+            let zelf = PyRef::<S>::try_from_object(vm, obj)?;
+            let value = V::try_from_object(vm, value)?;
+            (self)(&zelf, value, vm).into_pysetresult()
+        })
+    }
+}
+
+impl<F, T, V, R> IntoPySetterFunc<(OwnedParam<T>, V, R)> for F
+where
+    F: Fn(T, V) -> R + 'static,
+    T: TryFromObject,
+    V: TryFromObject,
+    R: IntoPySetResult,
+{
+    fn into_setter(self) -> PySetterFunc {
+        IntoPySetterFunc::into_setter(move |obj, v, _vm: &VirtualMachine| (self)(obj, v))
+    }
+}
+
+impl<F, S, V, R> IntoPySetterFunc<(RefParam<S>, V, R)> for F
+where
+    F: Fn(&S, V) -> R + 'static,
+    S: PyValue,
+    V: TryFromObject,
+    R: IntoPySetResult,
+{
+    fn into_setter(self) -> PySetterFunc {
+        IntoPySetterFunc::into_setter(move |zelf: &S, v, _vm: &VirtualMachine| (self)(zelf, v))
+    }
+}
+
+#[pyclass]
+pub struct PyGetSet {
+    name: String,
+    getter: Option<PyGetterFunc>,
+    setter: Option<PySetterFunc>,
+    // doc: Option<String>,
+}
+
+impl std::fmt::Debug for PyGetSet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "PyGetSet {{ name: {}, getter: {}, setter: {} }}",
+            self.name,
+            if self.getter.is_some() {
+                "Some"
+            } else {
+                "None"
+            },
+            if self.setter.is_some() {
+                "Some"
+            } else {
+                "None"
+            },
+        )
+    }
+}
+
+impl PyValue for PyGetSet {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.ctx.getset_type()
+    }
+}
+
+pub type PyGetSetRef = PyRef<PyGetSet>;
+
+impl PyBuiltinDescriptor for PyGetSet {
+    fn get(
+        zelf: PyRef<Self>,
+        obj: PyObjectRef,
+        _cls: OptionalArg<PyObjectRef>,
```

Comment:
Maybe to fix the exception.args case, you could have this check `cls` and if it's not there than just return `zelf`?

Reply:
I copied `descr_check` part of CPython. Now `descr_get` signature looks a little bit messy than before